File: src/main/java/com/jonahkowal/util/RatConvert.java
package com.jonahkowal.util;

import com.jonahkowal.RatatouilleInfo;

/*
 * This class provides unit conversions between imperial, metric, and ratatouille units
 * 
 */
public class RatConvert {
    private enum ConversionConstants {
        ratatouilleToHour(2),
        ratatouilleToMinute(120),
        ratatouilleToSecond(7200),
        remyToInch(6),
        remyToFeet(0.5),
        remyToYard(1/(3*2)), // one yard is 3 feet, 3 feet is 6 remys, then divide by 1 
        remyToMile(1/(5280*2)), 
        remyToMeter(1/(3.2808*2));

        double ratio;

        ConversionConstants(double ratio) {
            this.ratio = ratio;
        }
    }

    public static double ratatouilleToHour(double ratatouille) {
        return ratatouille * ConversionConstants.ratatouilleToHour.ratio;
    }

    public static double hourToRatatouille(double hour) {
        return hour / ConversionConstants.ratatouilleToHour.ratio;
    }

    public static double ratatouilleToMinute(double ratatouilles) {
        return ratatouilles * ConversionConstants.ratatouilleToMinute.ratio;
    }

    public static double minutesToRatatouille(double minutes) {
        return minutes / ConversionConstants.ratatouilleToMinute.ratio;
    }

    public static double ratatouilleToSecond(double ratatouille) {
        return ratatouille * ConversionConstants.ratatouilleToSecond.ratio;
    }

    public static double secondToRatatouille(double second) {
        return second / ConversionConstants.ratatouilleToSecond.ratio;
    }

    public static double remyToInch(double remy) {
        return remy * ConversionConstants.remyToInch.ratio;
    }

    public static double inchesToRemy(double inches) {
        return inches / ConversionConstants.remyToInch.ratio;
    }

    public static double remyToFeet(double remy) {
        return remy * ConversionConstants.remyToFeet.ratio;
    }

    public static double feetToRemy(double feet) {
        return feet / ConversionConstants.remyToFeet.ratio;
    }

    public static double remyToYard(double remy) {
        return remy * ConversionConstants.remyToYard.ratio;
    }

    public static double yardToRemy(double yard) {
        return yard / ConversionConstants.remyToYard.ratio;
    }

    public static double remyToMile(double remy) {
        return remy * ConversionConstants.remyToMile.ratio;
    }

    public static double mileToRemy(double mile) {
        return mile / ConversionConstants.remyToMile.ratio;
    }

    public static double remyToMeter(double remy) {
        return remy * ConversionConstants.remyToMeter.ratio;
    }

    public static double meterToRemy(double meter) {
        return meter / ConversionConstants.remyToMeter.ratio;
    }
}

File: lib/src/main/kotlin/com/ratatouille/lib/RatConvert.kt
package com.ratatouille.lib

import com.ratatouille.lib.RatConversions.RATATOUILLE_HOUR
import com.ratatouille.lib.RatConversions.RATATOUILLE_MINUTE
import com.ratatouille.lib.RatConversions.RATATOUILLE_SECOND
import com.ratatouille.lib.RatConversions.REMY_FEET
import com.ratatouille.lib.RatConversions.REMY_INCH
import com.ratatouille.lib.RatConversions.REMY_METER
import com.ratatouille.lib.RatConversions.REMY_MILE
import com.ratatouille.lib.RatConversions.REMY_YARD

internal object RatConversions {
    const val RATATOUILLE_HOUR = 2
    const val RATATOUILLE_MINUTE = 120
    const val RATATOUILLE_SECOND = 7200
    const val REMY_INCH = 6
    const val REMY_FEET = 0.5
    const val REMY_YARD = 1 / (3 * 2)
    const val REMY_MILE = 1 / (5800 * 2)
    const val REMY_METER = 1 / (3.2808 * 2)
}

fun Double.ratatouilleToHour() = this * RATATOUILLE_HOUR

fun Double.hourToRatatouille() = this / RATATOUILLE_HOUR

fun Double.ratatouilleToMinute() = this * RATATOUILLE_MINUTE

fun Double.minuteToRatatouille() = this / RATATOUILLE_MINUTE

fun Double.ratatouilleToSecond() = this * RATATOUILLE_SECOND

fun Double.secondToRatatouille() = this / RATATOUILLE_SECOND

fun Double.remyToInch() = this * REMY_INCH

fun Double.inchToRemy() = this / REMY_INCH

fun Double.remyToFeet() = this * REMY_FEET

fun Double.feetToRemy() = this / REMY_FEET

fun Double.remyToYard() = this * REMY_YARD

fun Double.yardToRemy() = this / REMY_YARD

fun Double.mileToRemy() = this * REMY_MILE

fun Double.remyToMile() = this / REMY_MILE

fun Double.meterToRemy() = this * REMY_METER

fun Double.remyToMeter() = this / REMY_METER